crates/sui-core/src/consensus_manager/mod.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use crate::authority::authority_per_epoch_store::AuthorityPerEpochStore;
+use crate::consensus_manager::mysticeti_manager::MysticetiManager;
+use crate::consensus_manager::narwhal_manager::{
+    NarwhalConfiguration, NarwhalManager, NarwhalManagerMetrics,
+};
+use fastcrypto::traits::KeyPair;
+use mysten_metrics::RegistryService;
+use narwhal_executor::ExecutionState;
+use narwhal_worker::TransactionValidator;
+use std::path::PathBuf;
+use std::sync::Arc;
+use sui_config::{ConsensusConfig, NodeConfig};
+
+pub mod mysticeti_manager;
+pub mod narwhal_manager;
+
+/// An enum to easily differentiate between the chosen consensus engine
+pub enum Manager {
```

Comment:
Have you considered using enum dispatch?

Reply:
Also, could we name it ConsensusManager instead for clarity?

Reply:
> Have you considered using enum dispatch?

haven't considered it indeed, but it's a great suggestion!

> Also, could we name it ConsensusManager instead for clarity?

ah originally had it ConsensusManger but then renamed it to Manager given that the module is already `consensus_manager` and afraid some overloading of meaning , but it seems that people are ok with it so can revert